docs/design/2020-05-08-standardize-error-codes-and-messages.md

```diff
@@ -162,35 +162,64 @@ In addition, the error codes should be append only in case of conflict between v
+- The error field: it's the error itself the user can view in TiDB system. Like `err.Error()`.
+- The description field: the description of the error, what happened and why happened? This could be written by developer outside the code, and the more detail this field explaining the better, even some guess of cause could be included.
+- The workaround filed: how to workaround this error.
 
 Besides, we can append a optional tags field to it:
 ```toml
 [error.9005]
 error = ""
-message = ""
+description = ""
 workaround = ""
 tags = ["tikv"]
 ```
 
-The tags is used to classify errors (e.g. the level of seriousness). At the very beginning, we can ignore it since we don't have enough errors listed. Once we have enough data, we need to classify all errors by different dimensions. Then we will make out a standard abount how to classify errors.
+The tags is used to classify errors (e.g. the level of seriousness). At the very beginning, we can ignore it since we don't have enough errors listed. Once we have enough data, we need to classify all errors by different dimensions. Then we will make out a standard about how to classify errors.
 
 #### The Error Code Range
 
+The error code is a 3-tuple of abbreviated component name, error class and error code, joined by a colon like `{Component}:{ErrorClass}:{ErrorCode}`.
+
+Where `Component` field is the abbreviated component name of the error source, wrote as upper case, component names are mapped as below:
+
+- TiKV: KV
+- PD: PD
+- DM: DM
+- BR: BR
+- TiCDC: CDC
+- Lightning: LN
+- TiFlash: FLASH
+- Dumpling: DP
+
+The `ErrorClass` is the name of the `ErrClass` the error belongs to, which defined by `errors.RegisterErrorClass` or someway likewise. If this is unacceptable (for projects not written with golang), anything that can classify the "type" of this error (e.g., package name.) would also be good. 
+
+The `ErrorCode` is the identity of this error. Both numeric and textual code are acceptable, but it would be better to provide textual code, which should be one or two short words with PascalCase named to identity the error.
+
+The content of `ErrorClass` and `ErrorCode` must matches `[0-9a-zA-Z]+`.
+
 For compatibility with MySQL protocol, the code transmitted through the mysql protocol should be number only, others can be a number with a prefix string.
 
 The code of each components looks like:
-- TiDB: [0, 9000), DB-([A-Z]+-)?[0-9]{3,}
-- TiKV: [9000, 9010), KV-([A-Z]+-)?[0-9]{3,}
-- PD: [9000, 9010), PD-([A-Z]+-)?[0-9]{3,}
-- DM: DM-([A-Z]+-)?[0-9]{3,}
-- BR: BR-([A-Z]+-)?[0-9]{3,}
-- CDC: CDC-([A-Z]+-)?[0-9]{3,}
-- Lightning: LN-([A-Z]+-)?[0-9]{3,}
-- Dumpling: DP-([A-Z]+-)?[0-9]{3,}
+
+```
+TiDB: {class}:{code}
+TiKV: KV:{class}:{code}
+PD: PD:{class}:{code}
+TiFlash: FLASH:{class}:{code}
+DM: DM:{class}:{code}
+BR: BR:{class}:{code}
+CDC: CDC:{class}:{code}
+Lightning: LN:{class}:{code}
+Dumpling: DP:{class}:{code}
+
+{class}, {code} ~= [A-Za-z0-9]+
+
+MySQL error code <-> TiDB error code
```

Comment:
please add more description about the relation between MySQL and TiDB error code